### PR TITLE
push tag latest

### PR DESCRIPTION
**Pull request by lwsanty**
have to merge https://github.com/athenianco/ci/pull/3 first

Signed-off-by: lwsanty <lwsanty@gmail.com>

### CODE REVIEW

**Comment by lwsanty** [comment]
@dpordomingo after ci [PR](https://github.com/athenianco/ci/pull/3) is merged it will work in such way:
```yaml
env:
  global:
    - GOOGLE_DOCKER_IMAGE=gcr.io/athenian-1/web-app
    - IMAGE_BRANCH_COMMIT=${GOOGLE_DOCKER_IMAGE}:${TRAVIS_BRANCH}-${TRAVIS_COMMIT:0:7}
...
script:
  # pushes branch-commit tag, sends it to the `version-controller`
  - IMAGE=${IMAGE_BRANCH_COMMIT} make deploy
  # pushes `latest`
  - IMAGE=${GOOGLE_DOCKER_IMAGE} docker-push
```

File: .travis.yml [hunk 1]
```diff
@@ -33,4 +34,5 @@ jobs:
     - stage: deploy
       name: 'Deploy App'
       script:
-        - IMAGE=${GOOGLE_DOCKER_IMAGE}:${TRAVIS_BRANCH}-${TRAVIS_COMMIT:0:7} make deploy
+        - IMAGE=${IMAGE_BRANCH_COMMIT} make deploy
+        - IMAGE=${GOOGLE_DOCKER_IMAGE} docker-push
```
Comment (se7entyse7en) [inline .travis.yml]:
```suggestion
        - IMAGE=${GOOGLE_DOCKER_IMAGE} make docker-push
```

**Comment by dpordomingo** [comment]
Thanks for your example @lwsanty.

What I don't understand is why doing:
```shell
$ IMAGE=gcr.io/athenian-1/web-app make docker-push
```
we are pushing `latest` tag, because it only calls:
```shell
$ gcloud docker -- push gcr.io/athenian-1/web-app
```

Because if I'm not wrong, this last one pushes all local `gcr.io/athenian-1/web-app` images into `gcr.io/athenian-1/web-app` remote registry (no mater the local tags we have, which can be error-prone); but the `latest` tag is not locally available at this point, because it was not tagged as far as I can see, because we only did:
```shell
$ IMAGE=gcr.io/athenian-1/web-app:master-901388f make deploy
```
which behaves as:
```shell
$ docker build -t gcr.io/athenian-1/web-app:master-901388f .
```
so it just build `master-901388f` tag, so `latest` is still missing.

**TL;DR**
imo, and following your example, we should do something like the following:

```yaml
env:
  global:
    - GOOGLE_DOCKER_IMAGE=gcr.io/athenian-1/web-app
    - IMAGE_BRANCH_COMMIT=${GOOGLE_DOCKER_IMAGE}:${TRAVIS_BRANCH}-${TRAVIS_COMMIT:0:7}
...
script:
  # pushes branch-commit tag, sends it to the `version-controller`
  - IMAGE=${IMAGE_BRANCH_COMMIT} make deploy
  # creates `latest`
  - docker tag ${IMAGE_BRANCH_COMMIT} ${GOOGLE_DOCKER_IMAGE}:latest
  # pushes `latest`
  - IMAGE=${GOOGLE_DOCKER_IMAGE}:latest docker-push
```

**Comment by lwsanty** [comment]
@dpordomingo you are right, I don't understand how I missed such an obvious thing, thank you!

File: .travis.yml [hunk 2]
```diff
@@ -33,4 +34,6 @@ jobs:
     - stage: deploy
       name: 'Deploy App'
       script:
-        - IMAGE=${GOOGLE_DOCKER_IMAGE}:${TRAVIS_BRANCH}-${TRAVIS_COMMIT:0:7} make deploy
+        - IMAGE=${IMAGE_BRANCH_COMMIT} make deploy
+        - docker tag ${IMAGE_BRANCH_COMMIT} ${GOOGLE_DOCKER_IMAGE}:latest
+        - IMAGE=${GOOGLE_DOCKER_IMAGE} make docker-push
```
Comment (dpordomingo) [inline .travis.yml]: What if we explicitly push the `latest` one, instead of all local `gcr.io/athenian-1/web-app` image tags?
```suggestion
        - IMAGE=${GOOGLE_DOCKER_IMAGE}:latest make docker-push
```
